cilium-operator: don't reconcile non-GAMMA xRoutes without a Cilium-managed Gateway

aetimmes opened the pull request:
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

<!-- Description of change -->

Fixes: #34754

```release-note
operator: don't reconcile non-GAMMA xRoutes without a Cilium-managed Gateway
```

sayboras commented on USERS.md:
```diff
@@ -639,9 +639,9 @@ Users (Alphabetically)
+      U: CNI, ClusterMesh, Network Policy, Hubble, Gateway API (GAMMA)
       L: https://seatgeek.com
-      Q: @byxorna
+      Q: @byxorna, @aetimmes
```
Thanks, can you split this change into separate PR to make it faster to review and merge?

sayboras commented on operator/pkg/gateway-api/httproute.go:
```diff
@@ -146,16 +146,28 @@ func (r *httpRouteReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		for _, parent := range hr.Spec.ParentRefs {
 			if helpers.IsGateway(parent) {
-				return true
+				// check if the Gateway has the correct controller
```
do we need to update similar logic for grpcroute and tlsroute ?

aetimmes replied:
likely yes - this logic also isn't currently working in my local setup; will revisit later today

aetimmes committed:
cilium-operator: don't reconcile non-Cilium xRoutes

Fixes: #34754

Currently, the cilium-operator xRoute reconcilers reconcile all
(non-GAMMA) xRoutes regardless of their parentage. This can lead to
race conditions where the Cilium operator can clobber metadata updates
made by other operatores (ie, the Kong Kubernetes Ingress Controller).
This PR adjusts the xRoute reconcilers to skip routes associated
with the Cilium gateway.

Signed-off-by: Andrew E. Timmes <andrew.timmes@gmail.com>
